### PR TITLE
net, e2e: Skip SR-IOV hotplug if FG is disabled

### PR DESCRIPTION
**Pull request by orelmisan**
PR #10185 introduced the SR-IOV migration-based hotplug with an E2E test that fails [1] in case the `HotplugNICs` feature gate is disabled.

The CI system of kubevirt/kubevirtci, has an SR-IOV lane that deploys a cluster and runs the conformance test mechanism with a focus on SR-IOV tests[2] without enabling the `HotplugNICs` feature gate.

This causes the pre-submit jobs in kubevirt/kubevirtci to fail[3], and requires a maintainer override in order to merge PRs.

As an immediate fix, skip the SR-IOV migration based hotplug tests if the `HotplugNICs` feature gate is disabled.

The kubevirt/kubevirt tests will not be affected by this change. A more permanent fix will follow.

[1] https://github.com/kubevirt/kubevirt/blob/cb29e9f87d41deae0d9e1efa6b0cde53ecd8c28a/tests/network/hotplug_sriov.go#L52
[2] https://github.com/kubevirt/project-infra/blob/88e8dbd315a7bc908f637ebe3168f1fe7da97839/github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml#L53
[3] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/1070/check-up-kind-1.27-sriov/1691706923780411392

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/network/hotplug_sriov.go [hunk 1]
```diff
@@ -49,7 +49,9 @@ var _ = SIGDescribe("[Serial] SRIOV nic-hotplug", Serial, decorators.SRIOV, func
 	sriovResourceName := readSRIOVResourceName()
 
 	BeforeEach(func() {
-		Expect(checks.HasFeature(virtconfig.HotplugNetworkIfacesGate)).To(BeTrue())
+		if checks.HasFeature(virtconfig.HotplugNetworkIfacesGate) {
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply tests/network/hotplug_sriov.go]: Thank you :)

File: tests/network/hotplug_sriov.go [hunk 2]
```diff
@@ -49,7 +49,9 @@ var _ = SIGDescribe("[Serial] SRIOV nic-hotplug", Serial, decorators.SRIOV, func
 	sriovResourceName := readSRIOVResourceName()
 
 	BeforeEach(func() {
-		Expect(checks.HasFeature(virtconfig.HotplugNetworkIfacesGate)).To(BeTrue())
+		if !checks.HasFeature(virtconfig.HotplugNetworkIfacesGate) {
+			Skip("HotplugNICs feature gate is disabled.")
+		}
```
Comment (oshoval) [inline tests/network/hotplug_sriov.go:54]: Make sense to move this please after the BeforeEach of line 57
(first to check setup even has SRIOV enabled)

Comment (orelmisan) [reply tests/network/hotplug_sriov.go:54]: I tend to agree, but I think it should be done in a follow-up PR.

Comment (oshoval) [reply tests/network/hotplug_sriov.go:54]: I think it is such small change, that even another commit fits, but whatever you prefer
We can save time of lanes, bump and so on, also review time

Comment (orelmisan) [reply tests/network/hotplug_sriov.go:54]: The suggested change will not affect the problem we want to quickly fix (PRs are blocked in kubevirt/kubevirtci).

Comment (oshoval) [reply tests/network/hotplug_sriov.go:54]: I know, but my point is different, it is a cosmetic logical that can be done in a jiffy, and can save resources
but from here it is yours

**Review by oshoval — APPROVED** [review]

File: tests/network/hotplug_sriov.go (hunk 2 above)
Comment (orelmisan) [reply tests/network/hotplug_sriov.go:54]: It is approved.

Comment (oshoval) [reply tests/network/hotplug_sriov.go:54]: a change doesnt clean the approve, this what i meant

**Comment by orelmisan** [comment]
Thank you @oshoval

**Pull request merged**